### DISCUSSION TITLE
Database transitions

### DISCUSSION THREAD

**Discussion by yashdesai87** [spatie/laravel-model-states#264] [Q&A]
is there a way to define the transitions in a database than putting them explicitly in a state file?

We are building a drag and drop transition creator and these transitions need to be stored in the database.

**Comment by jonjakoblich** [comment]
You can provide allowed transitions from the current state via the `transitionableStates()` method to your frontend. Example seen in the docs: https://spatie.be/docs/laravel-model-states/v2/working-with-transitions/04-retrieving-transitionable-states

**Reply by Norda-AD** [reply]
My guess is the question is about defining the state machine in a database instead of the file. This way you could offer different workflow depending on the user. Not all user need the same transition nor states. A little bit like in Jira or other tools where you can configure the workflow depending on your process.
